File: physicslayout/src/main/java/com/jawnnypoo/physicslayout/PhysicsLayoutParams.java
package com.jawnnypoo.physicslayout;

/**
 * Signifies that the LayoutParams are able to provide a PhysicsConfig. You still need to create a
 * {@link android.view.ViewGroup.LayoutParams} implementation using the appropriate subclass, and
 * making use of {@link PhysicsLayoutParamsProcessor}
 */
public interface PhysicsLayoutParams {
    PhysicsConfig getConfig();
}

File: physicslayout/src/main/java/com/jawnnypoo/physicslayout/PhysicsLayoutParams.kt
package com.jawnnypoo.physicslayout

/**
 * Signifies that the LayoutParams are able to provide a PhysicsConfig. You still need to create a
 * [android.view.ViewGroup.LayoutParams] implementation using the appropriate subclass, and
 * making use of [PhysicsLayoutParamsProcessor]. See [PhysicsFrameLayout] for an example
 */
interface PhysicsLayoutParams {
    val config: PhysicsConfig
}
